it/api/GetIncomeSourcesITest.scala

```diff
@@ -70,25 +95,72 @@ class GetIncomeSourcesITest extends PlaySpec with WiremockSpec with ScalaFutures
 
       "return 503 if a downstream error occurs" in new Setup {
         val responseBody = "{\"code\":\"SERVICE_UNAVAILABLE\",\"reason\":\"The service is temporarily unavailable\"}"
-        stubGetWithResponseBody(s"/income-tax-dividends/income-tax/nino/AA123123A/sources\\?taxYear=2019&mtditid=123123123", SERVICE_UNAVAILABLE, responseBody)
-        stubGetWithResponseBody(s"/income-tax-interest/income-tax/nino/AA123123A/sources\\?taxYear=2019&mtditid=123123123", SERVICE_UNAVAILABLE, responseBody)
+        stubGetWithResponseBody(
+          url = s"/income-tax-dividends/income-tax/nino/AA123123A/sources\\?taxYear=2019",
+          status = SERVICE_UNAVAILABLE,
+          response = responseBody,
+          requestHeaders
+        )
+
+        stubGetWithResponseBody(
+          url = s"/income-tax-interest/income-tax/nino/AA123123A/sources\\?taxYear=2019",
+          status = SERVICE_UNAVAILABLE,
+          response = responseBody,
+          requestHeaders
+        )
         authorised()
 
         whenReady(buildClient(s"/income-tax-submission-service/income-tax/nino/$successNino/sources")
-          .withQueryStringParameters("taxYear" -> "2019", "mtditid" -> "123123123").get) {
+          .withQueryStringParameters("taxYear" -> "2019")
+          .withHttpHeaders(mtditidHeader)
+          .get) {
           result =>
             result.status mustBe 503
             result.body mustBe "{\"code\":\"SERVICE_UNAVAILABLE\",\"reason\":\"The service is temporarily unavailable\"}"
         }
       }
 
       "return 401 if the user has no HMRC-MTD-IT enrolment" in new Setup {
-        stubGetWithoutResponseBody(s"/income-tax-dividends/income-tax/nino/AA123123A/sources\\?taxYear=2019&mtditid=123123123", SERVICE_UNAVAILABLE)
-        stubGetWithoutResponseBody(s"/income-tax-interest/income-tax/nino/AA123123A/sources\\?taxYear=2019&mtditid=123123123", SERVICE_UNAVAILABLE)
+        stubGetWithoutResponseBody(
+          url = s"/income-tax-dividends/income-tax/nino/AA123123A/sources\\?taxYear=2019&mtditid=123123123",
+          status = SERVICE_UNAVAILABLE,
+          requestHeaders
+        )
+
+        stubGetWithoutResponseBody(
+          url = s"/income-tax-interest/income-tax/nino/AA123123A/sources\\?taxYear=2019&mtditid=123123123",
+          status = SERVICE_UNAVAILABLE,
+          requestHeaders
+        )
+        unauthorisedOtherEnrolment()
+
+        whenReady(buildClient(s"/income-tax-submission-service/income-tax/nino/$successNino/sources")
+          .withQueryStringParameters("taxYear" -> "2019")
+          .withHttpHeaders(mtditidHeader)
+          .get) {
+          result =>
+            result.status mustBe 401
+            result.body mustBe ""
+        }
+      }
+
+      "return 401 if the request has no MTDITID header present" in new Setup {
+        stubGetWithoutResponseBody(
+          url = s"/income-tax-dividends/income-tax/nino/AA123123A/sources\\?taxYear=2019&mtditid=123123123",
```

Comment:
These stubs can be removed because this request would fail fast in the AuthorisedAction so the stubs would never be called.